### PR TITLE
Migrating `google_client_config` & `google_client_openid_userinfo` data sources to plugin framework

### PR DESCRIPTION
**Pull request by SarahFrench**
This PR migrates these data sources to the plugin-framework side of the muxed provider:
- https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/client_config 
- https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/client_openid_userinfo

---

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: mmv1/third_party/terraform/utils/framework_provider.go.erb [hunk 1]
```diff
@@ -37,17 +38,19 @@ func New(version string) provider.ProviderWithMetaSchema {
...
     gRPCLoggingOptions []option.ClientOption
     pollInterval time.Duration
     project types.String
-    region string
+    region types.String
+    zone types.String
     requestBatcherIam *RequestBatcher
     requestBatcherServiceUsage *RequestBatcher
     scopes []string
+    tokenSource oauth2.TokenSource
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/utils/framework_provider.go.erb:53]: This follows the example of : https://github.com/hashicorp/magic-modules/blob/19b676e971d55e8524600bb33ae62a10d43b223c/mmv1/third_party/terraform/utils/config.go.erb#L194

**Comment by SarahFrench** [comment]
👆 I think the reason for my second load of tests not cancelling the first job is that I force-pushed to rebase the branch and so the commits didn't match

File: mmv1/third_party/terraform/data_sources/data_source_google_client_config.go [hunk 2]
```diff
@@ -1,60 +1,126 @@
...
+func (d *GoogleClientConfigDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 
-			"zone": {
-				Type:     schema.TypeString,
+	resp.Schema = schema.Schema{
+
+		Description:         "Use this data source to access the configuration of the Google Cloud provider.",
+		MarkdownDescription: "Use this data source to access the configuration of the Google Cloud provider.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
```
Comment (megan07) [inline mmv1/third_party/terraform/data_sources/data_source_google_client_config.go:48]: Can we add descriptions to the id too? for consistency

**Review by megan07 — APPROVED** [review]
One minor comment about adding descriptions to `id` fields, but other than that this looks good to me! Thanks!

**Comment by SarahFrench** [comment]
Failed test isn't related to this PR; merging!

**Pull request merged**